### PR TITLE
support AMP

### PR DESCRIPTION
**Pull request by lzx1413**
1. support AMP training from [pytorch >= 1.6](https://pytorch.org/docs/stable/notes/amp_examples.html)
2. update focalloss with d2 version
3. update ctr loss with BCELoss
4. add amp train example config which can reduce about 50% of the GPU mem usage

### CODE REVIEW

File: videoanalyst/model/loss/loss_impl/sigmoid_ce_centerness.py [hunk 1]
```diff
@@ -56,13 +56,8 @@ def forward(self, pred_data, target_data):
         pred = pred_data["ctr_pred"]
         label = target_data["ctr_gt"]
         mask = (~(label == self.background)).type(torch.Tensor).to(pred.device)
-        not_neg_mask = (pred >= 0).type(torch.Tensor).to(pred.device)
-        loss = (pred * not_neg_mask - pred * label +
-                self.safelog(1. + torch.exp(-torch.abs(pred)))) * mask
-        loss_residual = (-label * self.safelog(label) -
-                         (1 - label) * self.safelog(1 - label)
-                         ) * mask  # suppress loss residual (original vers.)
-        loss = loss - loss_residual.detach()
```
Comment (MARMOTatZJU) [inline videoanalyst/model/loss/loss_impl/sigmoid_ce_centerness.py:65]: Need to keep residual subtraction so that we can see the reduction of centerness loss more clearly
